staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time_test.go

```diff
@@ -171,3 +171,20 @@ func TestTimeProto(t *testing.T) {
 		}
 	}
 }
+
+func TestMarshalQueryParameterNilPointer(t *testing.T) {
+	var x *Time
+	v, err := x.MarshalQueryParameter()
+	if err != nil || v != "" {
+		t.Errorf("Unexpected results: '%s', %v", v, err)
+	}
+}
+
+func TestMarshalJSONNilPointer(t *testing.T) {
+	t.Skip("not sure if this should be supported")
```

Comment:
I've removed the `MarshalJSON` test to reduce the scope of this PR. Hope it will help to get this merged faster...